Automattic/wp-calypso
# SSR: Use document-head to populate `title`, `meta`s, and `link`s; drop `Helmet`

This PR:

* has the theme sheet and showcase use `DocumentHead` instead of `Head` (which was a rather specialized component tailored to the theme showcase's SEO needs, based on `react-helmet``
* drops `Head`, and the `react-helmet` dependency
* modifies `server/render` to use data from the `document-head` state to populate SEO relevant fields (`title`, `meta`, `link`) for SSR
* removes `description` related reducer and actions.
 * Description meta can be handled by just using `addMeta`, so I don't think it's worthwhile having specialized functions around.

To test:
* While logged out, check a given theme sheet's page source. Verify that SEO relevant tags are present (`title`, `meta`, `link`). (Compare with `master` or `production`.)
* While logged out, navigate to the showcase (http://calypso.localhost:3000/design), and click on any theme. Watch the title change -- at no point should it not make any sense.
* Same when logged in. (Also, there obviously shouldn't be any errors, console or otherwise.)

TODO (separate PR):
* Make sure we SSR `title` et al for `/design` (_this isn't currently done on `master`, so not a regression_)

Fixes #3795

Test live: https://calypso.live/?branch=remove/head
